.github/workflows/integration-test-trigger.yml

```diff
@@ -2,6 +2,8 @@ name: "Integration test (trigger only)"
 
 on:
   push:
+    branches: [main]
+  pull_request:
```

Comment:
Do you want to filter this to PRs against `main`?

Reply:
I think we want to run against all PRs. That gives more immediate feedback even if you're merging into another feature branch. Shouldn't make too much difference in practice. The `integration-tests-have-been-run` is also not a required check currently.